Xamarin.Forms.Platform.Tizen/Renderers/SwipeViewRenderer.cs

```diff
@@ -0,0 +1,435 @@
+
+		int MaximumSwipeSize { get; set; }
+
+		bool IsHorizontalSwipe => SwipeDirection == SwipeDirection.Left || SwipeDirection == SwipeDirection.Right;
+		bool IsNegativeDirection => SwipeDirection == SwipeDirection.Left || SwipeDirection == SwipeDirection.Up;
+
+		SwipeItems CurrentItems { get; set; }
+
+
+		protected override void OnElementChanged(ElementChangedEventArgs<Layout> e)
+		{
+			base.OnElementChanged(e);
+			Initialize();
+		}
+
+		protected override void Dispose(bool disposing)
+		{
+			if (disposing)
+			{
+				_gestureLayer?.Unrealize();
+				_gestureLayer = null;
+				_itemsRenderer?.Dispose();
+				_itemsRenderer = null;
+			}
+			base.Dispose(disposing);
+		}
+
+		void Initialize()
+		{
+			_gestureLayer?.Unrealize();
+			_gestureLayer = new GestureLayer(NativeView);
+			_gestureLayer.Attach(NativeView);
+
+			_gestureLayer.SetMomentumCallback(GestureLayer.GestureState.Start, OnStarted);
+			_gestureLayer.SetMomentumCallback(GestureLayer.GestureState.Move, OnMoved);
+			_gestureLayer.SetMomentumCallback(GestureLayer.GestureState.End, OnEnd);
+			_gestureLayer.SetMomentumCallback(GestureLayer.GestureState.Abort, OnEnd);
+			SwipeDirection = 0;
+			DrawerState = SwipeDrawerState.Closed;
+			Control.AllowFocus(true);
+			Control.Unfocused += (s, e) =>
+			{
+				_ = SwipeClose();
+			};
+		}
+
+		void OnStarted(GestureLayer.MomentumData moment)
+		{
+		}
+
+		void OnMoved(GestureLayer.MomentumData moment)
+		{
+			if (SwipeDirection == 0)
+			{
+				var direction = SwipeDirectionHelper.GetSwipeDirection(new Point(moment.X1, moment.Y1), new Point(moment.X2, moment.Y2));
+
+				if (HasRightItems && direction == SwipeDirection.Left)
+				{
+					SwipeDirection = SwipeDirection.Left;
+				}
+				else if (HasLeftItems && direction == SwipeDirection.Right)
+				{
+					SwipeDirection = SwipeDirection.Right;
+				}
+				else if (HasTopItems && direction == SwipeDirection.Down)
+				{
+					SwipeDirection = SwipeDirection.Down;
+				}
+				else if (HasBottomItems && direction == SwipeDirection.Up)
+				{
+					SwipeDirection = SwipeDirection.Up;
+				}
+				else
+				{
+					return;
+				}
+
+				UpdateItems();
+				SwipeView.SendSwipeStarted(new SwipeView.SwipeStartedEventArgs(SwipeDirection));
+			}
+
+			var offset = GetSwipeOffset(moment);
+
+			if (IsNegativeDirection)
+			{
+				if (offset > 0)
+					offset = 0;
+			}
+			else
+			{
+				if (offset < 0)
+					offset = 0;
+			}
+
+			if (Math.Abs(offset) > MaximumSwipeSize)
+			{
+				offset = MaximumSwipeSize * (offset < 0 ? -1 : 1);
+			}
+
+			var toDragBound = NativeView.Geometry;
+			if (IsHorizontalSwipe)
+			{
+				toDragBound.X += offset;
+			}
+			else
+			{
+				toDragBound.Y += offset;
+			}
+			Platform.GetRenderer(SwipeView.Content).NativeView.Geometry = toDragBound;
+			SwipeView.SendSwipeChanging(new SwipeView.SwipeChangingEventArgs(SwipeDirection, Forms.ConvertToScaledDP(offset)));
+		}
+		async void OnEnd(GestureLayer.MomentumData moment)
```

Comment:
The name of an async method, by convention, ends with an "Async" suffix.  (same below)

Reply:
I don't think so, because it return `void` type. `async` is just for internal implement (it is not `API`), 
if internal implementation was changed, `async` could be removed.